facefusion/job_runner.py

```diff
@@ -0,0 +1,97 @@
+import os
+import shutil
+from typing import Dict
+from facefusion.ffmpeg import concat_video
+from facefusion.filesystem import is_video, create_temp, get_temp_file_path, is_directory, is_file
+from facefusion import logger, wording
+from facefusion.job_manager import read_job_file, set_step_status, move_job_file, get_job_ids, get_total_steps, get_step_status
+from facefusion.typing import JobStep, HandleStep
+
+
+def run_jobs(handle_step : HandleStep) -> None:
+	job_queued_ids = sorted(get_job_ids('queued'))
+
+	for job_id in job_queued_ids:
+		run_job(job_id, handle_step)
+		total_steps = get_total_steps(job_id)
+		completed_steps = 0
+
+		for step_index in range(total_steps):
```

Comment:
we need a listing for completed, failed and queue steps anyway for the ui... try to make a general solution instead of this for()